### DISCUSSION TITLE
Need help with labwc on raspbian OS lite (Bookworm)

### DISCUSSION THREAD

**Discussion by noobmaster777** [labwc/labwc#2385] [Q&A]
I am so lost. My goal is to run a terminal emulator on labwc or a mini desktop whenever I run labwc. So I downloaded labwc via apt and copied the configurations on the labwc website but whenever I enter labwc, my tty just stops, so I move to tty2, I dont find anything on journalctl. Im on a raspberry pi 5 with a 3.5inch tft screen. Can someone guide me on the basics and the required apps on how I can do this? Apologies and thanks in advance!

**Comment by johanmalm** [comment]
It's not clear to me what's going on, but it's possible that you've started labwc but not loaded any wallpaper clients, panels, and so on.

Does it work if you run `labwc-pi` (`/usr/bin/labwc-pi`) or simply `labwc -m`?

IIRC the pi uses `pcmanfm --desktop` for wallpaper and `wf-panel-pi` for panel if that's any help.

**Reply by noobmaster777** [reply]
There is no labwc-pi coming from raspberry pi os lite. Thanks for the tips, will do testing soon then update you!

**Reply by noobmaster777** [reply]
Downloaded but no difference. I also tried downloading sway and waybar for the example autostart file found on the labwc website and placed in .config/labwc/autostart but it still does the same. Can I have your autostart for pcmanfm with the wl-panel pi? labwc --debug > labwc_debug.log indicates that render/egl.c is INVALID and an error on  [libseat] [libseat/backend/logind.c could not close device. Device not taken. [ERROR] [backend/session/session.c] failed to close device. Resource temporarily unavailable

**Reply by tokyo4j** [reply]
Maybe you need to install polkit.

**Reply by noobmaster777** [reply]
> Maybe you need to install polkit.

It was already installed. Is there anything i should do to set it up?

**Comment by cillian64** [comment]
Could you try installing `raspberrypi-ui-mods`?  That will install the default rpi labwc config and the `labwc-pi` launcher so that everything should just work.  You can use the `--no-install-recommends` argument to `apt-get` to cut down on the extra packages if that's an issue.

**Reply by noobmaster777** [reply]
> Could you try installing `raspberrypi-ui-mods`? That will install the default rpi labwc config and the `labwc-pi` launcher so that everything should just work. You can use the `--no-install-recommends` argument to `apt-get` to cut down on the extra packages if that's an issue.

Purged labwc sway waybar mako-notifier then autopurged and then proceeded to download raspberrypi-ui-mods with apt. Typed in labwc-pi but still does the same. Upon debugging new errors have occurred. 

1) [ERROR] [EGL] command: eglQueryDeviceStringEXT, error: EGL_BAD_PARAMETER (0x300c), message: "eglQueryDeviceStringEXT"

2) (light-locker:1149) ERROR: Environment variable XDG_SESSION_PATH not set. Is LightDM running?

3) (polkit-mate-authentication-agent-1:1144): GLib-CRITICAL: g_variant_new_string: assertion 'string != NULL' failed
(polkit-mate-authentication-agent-1:1144): polkit-mate-1-WARNING: Failed to register client: GDBus.Error:  ....

4) failed to load info for lxde-x-www-browser.desktop
Failed to load info for lxterminal.desktop
Could not open plugin wf-panel-pi/libconnect.so
iw: not found
Wlr-randr not found

**Reply by Consolatis** [reply]
This looks like the depends in the raspberrypi-ui-mods package are not up to date.

**Reply by spl237** [reply]
ui-mods is intended to be used as part of the image creation process - it was never intended to be a method for converting a non-desktop image to a desktop image. I don't think changing its depends will make much difference - the OP is simply trying to do something we have never supported. Our position has always been that the lite image is for people who don't want a desktop; the desktop image is for people who do. Trying to convert one to the other is not something we support, or have ever put any effort into making work!

**Comment by johanmalm** [comment]
Could you try deleting (renaming) $HOME/.config/labwc/

Then run with labwc-pi

(Just to rule out that you haven’t got anything in the home dir that clashes with the rpi configs)

**Reply by noobmaster777** [reply]
Ok just did that but same errors.
Possible that im lacking xdg configs on /etc/environment? If I type sudo labwc-pi i only get one error. XDG_RUNTIME_DIR is unset

**Comment by spl237** [comment]
Did you start from a lite image, and are now trying to add a desktop to it?

That's not something we support or recommend, and is unlikely to ever work.

I'd recommend starting with a desktop image - which should already be working and starting labwc - and then starting your desired terminal emulator using a desktop file in the ~/.config/autostart directory.

**Reply by noobmaster777** [reply]
Yes. I thought it was possible. My bad

**Reply by noobmaster777** [reply]
> Did you start from a lite image, and are now trying to add a desktop to it?
> 
> That's not something we support or recommend, and is unlikely to ever work.
> 
> I'd recommend starting with a desktop image - which should already be working and starting labwc - and then starting your desired terminal emulator using a desktop file in the ~/.config/autostart directory.

So I tried this. Fresh raspberry pi OS 64bit (1.2GB) image from raspberry pi Imager and proceeded to make a labwc desktop environment work. In my config.txt i put in dtoverlay=waveshare35a so my tft would work. At boot, it goes to a blackscreen with a blinking cursor on the upper left, but only up to there. I can still do ctrl+alt+F2 to go to another tty. Can you help me figure this out?

**Reply by cillian64** [reply]
Just to check, you're seeing the black screen with blinking cursor and also are able to use TTY2 on the TFT display?

Please could you run the following from TTY2 to capture debug logs then attach the contents of `labwc_log` here?

```
sudo systemctl stop lightdm
labwc-pi &>labwc_log
```

**Reply by noobmaster777** [reply]
Thanks for the help. I moved back to rpi bookworm lite and closing this.
